### ISSUE TITLE
Implementing Delete Account and Delete Team

### ISSUE THREAD

**Issue by hakankaradis** [koding/koding#9131]
This discussion is for specifying all requirements and user flow.

We need to design user flow for deleting account and deleting team.

As a member what will happen if I delete my account
As an admin what will happen if I delete my account (what if admin created the team stack)
As an owner what will happen if I delete my account or delete my team.

## Requirements

When member wants to delete his/her account
- We will remove him from team and delete his account.
- Will Admins see his team stack on the sidebar (like he is disabled)

When admin wants to delete his/her account
- What if he has stack that is shared with team.
- What if he has credentials that is shared with team.
- Does admin has right to delete team?

When owner wants to delete his/her account.
- Are we going to delete team as well?
- Are we going to assign team ownership to admin?
- Are we going to ask to owner to set his ownership to any admin? (what If he has no admin in his team)

When owner delete his/her team
- What will happen to other team members? (Notification)

### Member
As a member I click to delete account
- Ask for confirmation
- Leave team
- Assign his team stack to admins
- Notify Admins

### Admin
As an admin I click to delete account
- Ask for confirmation
- Leave team
- Assign his team stack to admins
- Notify Admins

### Owner
As an owner I click to delete account
- Is there any admin or member in the team? 
  - If so, ask owner to assign ownership to any admin or member
  - If not, delete account

As an owner I click to delete team
- Will we notify the members? (maybe by email)
  
I have tried to create an outline for these issues. Lets discuss how can we improve and specify all requirements and user flows. What we need for back-end and front-end?
/cc: @gokmen @cihangir @sibblegp @sinan @usirin 

We will also need UI for these features
/cc: @ftasdelen

**Comment by hakankaradis** [comment]
These issues has pre-requirement.
We need to implement a `Team Resource Management` page to give admins chance to edit/delete team resources. It is a blocker right now.

@ftasdelen Can you please share your design for this page?

**Comment by ftasdelen** [comment]
The latest designs are here.
[Resources.pdf](https://github.com/koding/koding/files/499013/Resources.pdf)

I think we don't have "delete all" functionality yet. Lets discuss that.

**Comment by sibblegp** [comment]
I agree with most of those features, but can stacks be assigned to more than one person?  You say "Assign his team stack to admins".  Doesn't a stack belong to a single user?  And why not have it go to the owner?

**Comment by hakankaradis** [comment]
As you said stack belongs to a single user. For example, when an admin disable any member, his stack will be assigned to admin who disabled the member. So when owner wants to delete his team, he will need to manage all stacks in team so that's why we need Resource Management page to not have that kind of problem. We will ask owner to delete all stacks that are belong to team. I believe it will solve the problem

**Comment by sibblegp** [comment]
So when a member deletes their stacks, how do we decide which admin the stacks will get assigned to?  Or should we just assign them to the owner (my preference)?

**Comment by hakankaradis** [comment]
Member can only delete his private stack, so we don't need to assign the stacks that are private to anyone

**Comment by sibblegp** [comment]
I meant the team stacks.  Which ones will they get assigned to as the owner?
